### PR TITLE
Precursor to allow credentials from Jenkins Credentials

### PR DESCRIPTION
**Pull request by MattLud**
This work is needed to support passing credentials from Jenkins to the gerrit library.

### CODE REVIEW

File: src/main/java/com/sonymobile/tools/gerrit/gerritevents/ssh/SshConnectionImpl.java [hunk 1]
```diff
@@ -400,6 +400,7 @@ public synchronized void disconnect() {
             logger.debug("Disconnecting client connection.");
             connectSession.disconnect();
             connectSession = null;
+            authentication.deleteTempKeyFile();
```
Comment (rsandell) [inline src/main/java/com/sonymobile/tools/gerrit/gerritevents/ssh/SshConnectionImpl.java]: Isn't it a bit dangerous to have the key lying around on the file system for that long? Especially in a temp folder?

**Comment by rsandell** [comment]
You can probably somewhat easy make a for example `StringIdentity` class and add that via [JSch.addIdentity(Identity identity, byte[] passphrase)](https://epaul.github.io/jsch-documentation/javadoc/com/jcraft/jsch/JSch.html#addIdentity-com.jcraft.jsch.Identity-byte:A-) instead of writing the file to disk.

Look at for example how `com.jcraft.jsch.IdentityFile.newInstance` is implemented for inspiration.

File: src/main/java/com/sonymobile/tools/gerrit/gerritevents/ssh/SshConnectionImpl.java (hunk 1 above)
Comment (MattLud) [reply src/main/java/com/sonymobile/tools/gerrit/gerritevents/ssh/SshConnectionImpl.java]: Agreed - it was designed to provide compatibility on with the existing behavior(read a key from disk). The key could be written to user home with limited readability. There's also no reason it has to stay this way per your other comment.

**Comment by clabu** [comment]
This is two years old. Is this pull request still needed, or has this use case been solved by some other code change?

**Comment by rsandell** [comment]
Might still be needed, but PRs can be revived.

**Pull request closed without merging**